### DISCUSSION TITLE
Passing `undefined` to a component property should not override theme styles

### DISCUSSION THREAD

**Discussion by Nantris** [react-native-elements/react-native-elements#3640] [Ideas]
Maybe I'm overlooking some things this makes easier, but in my experience it only makes things harder.

**Comment by arpitBhalla** [comment]
Can you elaborate it with example, so it'd be more clear to us?

**Reply by Nantris** [reply]
Sure.  If you set a `backgroundColor` of red on the `CheckBox`'s `containerStyle` via `ThemeProvider`, that sets the background color to red as expected.

Now if you write some code like this, I'd expect that the background color would still be red if `someProp` is `false`, but instead it becomes transparent (or at least looks that way) because `undefined` overrides the theme level properties.

```js
  const backgroundColor = someProp ? 'green' : undefined;
  return (
    <CheckBox containerStyle={{ 
       backgroundColor
     }} />
   )
```
